[tools/depends] bump python 3.9.7

fuzzard opened the pull request:
## Description
Update python to 3.9.7 along with all dependencies required

## Motivation and context
Need >= python 3.9.1 for native apple silicon support.
Also have been trying to bump python for most of the v20 cycle without much success due to trying to transition to a different build mechanism. For now, lets just get newer python in, and i'll continue to work on the build system when im not trying to introduce huge version bumps.

Python 3.9 has a new dependency of autoconf-archive
Bumped meson to a version that supports python 3.9
All patches fixed up to match 3.9.7

We are carrying an additional patch for android python https://github.com/xbmc/xbmc/commit/c3acaecf7989c136a5402b7628e9a39f33519a9b
This is cherry-picked from cpython upstream, and was merged after the 3.9.7 release. Will be in 3.9.8 (and of course 3.10+), so can drop it then.

@wsnipex if you wouldnt mind throwing autocont-archive and meson on mirrors at your convenience.
@Paxxi if you want to bump windows, let me know. Not fussed either way, and i dont intend on holding this up if you dont have time, so all good either way.

RE python api changes. As far as im aware theres nothing critical needing to be changed in LanguageInvoker at the moment. Have been meaning to look at https://docs.python.org/3/c-api/init_config.html#init-config to convert us over to that, but not necessary right now.
3.10 will bring a far few deprecations and removals, so figure might be safe to not jump ahead to that at this stage.

## How has this been tested?
Runtime tested ios/macos. Tested against PIL and pycryptodome test addons, still function.
Needs android testing, particularly around the binary module usage (PIL, pycryptodome)

## What is the effect on users?
Newer python version

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

Commit:
[tools/depends][native] implement autoconf-archive

python 3.9+ has a dependency on autoconf-archive

Commit:
[tools/depends][native] bump python to 3.9.7

Commit:
[tools/depends][native] bump meson 0.59.0

Commit:
[tools/depends][target] python 3.9.7

Bump to python 3.9.7

Commit:
[tools/depends][target] python3 Move crypt.h include from public header

backport https://github.com/python/cpython/pull/27394 to 3.9.6
Android Pycryptodome builds fail due to crypt.h inclusion (not respecting undef HAVE_CRYPT_H)

Commit:
[tools/depends][target][pythonmoduile-pycryptodome] ios/tvos fix xcode 11 failure

xcode 11 (eg jenkins) strips out the existing -arch flag when building module
Xcode 12 correctly passes through our cflags unadulterated. Work around the xcode 11
failure by explicitly adding -arch triplet that doesnt get stripped

Commit:
[tools/depends][target][pythonmoduile-pil] ios/tvos fix xcode 11 failure

xcode 11 (eg jenkins) strips out the existing -arch flag when building module
Xcode 12 correctly passes through our cflags unadulterated. Work around the xcode 11
failure by explicitly adding -arch triplet that doesnt get stripped

python_o is also not required, so remove

ksooo reviewed (commented):
You need to change this as well, to not break caching of compiled python modules on Android: https://github.com/xbmc/xbmc/blob/master/xbmc/platform/android/activity/XBMCApp.cpp#L1405

Commit:
[compileinfo] add python version

Instead of hardcoding python versions, use compileinfo to inject version numbers
used to compile kodi

fuzzard commented:
Going to merge this in a couple days if no further comments

fuzzard commented:
Jenkins build and merge
